GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -294,6 +294,18 @@ public EventMetadata GetAndResetHeartBeatMetadata(out bool logToFile)
             return metadata;
         }
 
+        public bool DehydrateFolder(string relativePath)
+        {
+            // Remove all modified paths that start with the path
+            this.modifiedPaths.RemoveAllEntriesForFolder(relativePath);
+
+            // Remove all placeholders that start with the path
+            this.placeholderDatabase.RemoveStartingWith(relativePath);

```

Comment:
nit: Maybe call tihs RemoveAllEntriesForFolder as well?

Reply:
That is not what happens from the perspective of the `placeholderDatabase`. It is using a `LIKE 'path%`.  You could pass a partial folder name and it would remove all the entries that start with that partial folder name.  It doesn't know about folders only paths.